From 8e78ffc54b2a5444e96cdcc1440928b04e21c2a5 Mon Sep 17 00:00:00 2001
From: AppaRao Puli <apparao.puli@intel.com>
Date: Thu, 6 Jan 2022 01:37:08 +0000
Subject: [PATCH] Convert VariantType to DbusVariantType

All bmcweb code is now converted to use
DbusVariantType to reduce the image size.

Its missed in one place where the code is under compiler
flag BMCWEB_ENABLE_REDFISH_PROVISIONING_FEATURE. This commit
convert missed types to DbusVariantType.

Tested:
After conversion, image builds fine with compiler
flag enabled. Also tested the PFR provisioned
dbus calls and it works fine.

Signed-off-by: AppaRao Puli <apparao.puli@intel.com>
---
 redfish-core/lib/systems.hpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/redfish-core/lib/systems.hpp b/redfish-core/lib/systems.hpp
index 5fbd568..e9ae7ca 100644
--- a/redfish-core/lib/systems.hpp
+++ b/redfish-core/lib/systems.hpp
@@ -1840,7 +1840,8 @@ inline void getProvisioningStatus(std::shared_ptr<bmcweb::AsyncResp> aResp)
     BMCWEB_LOG_DEBUG << "Get OEM information.";
     crow::connections::systemBus->async_method_call(
         [aResp](const boost::system::error_code ec,
-                const std::vector<std::pair<std::string, VariantType>>&
+                const std::vector<
+                    std::pair<std::string, dbus::utility::DbusVariantType>>&
                     propertiesList) {
             nlohmann::json& oemPFR =
                 aResp->res.jsonValue["Oem"]["OpenBmc"]["FirmwareProvisioning"];
@@ -1858,8 +1859,8 @@ inline void getProvisioningStatus(std::shared_ptr<bmcweb::AsyncResp> aResp)
 
             const bool* provState = nullptr;
             const bool* lockState = nullptr;
-            for (const std::pair<std::string, VariantType>& property :
-                 propertiesList)
+            for (const std::pair<std::string, dbus::utility::DbusVariantType>&
+                     property : propertiesList)
             {
                 if (property.first == "UfmProvisioned")
                 {
-- 
2.17.1

